Sources/Plasma/PubUtilLib/plClipboard/plClipboard.cpp

```diff
@@ -89,29 +91,38 @@ void plClipboard::SetClipboardText(const ST::string& text)
 {
     if (text.empty())
         return;
+
 #ifdef HS_BUILD_FOR_WIN32
     ST::wchar_buffer buf = text.to_wchar();
     size_t len = buf.size();
 
-    if (len == 0) 
+    if (len == 0)
         return;
 
-    std::unique_ptr<void, HGLOBAL(WINAPI*)(HGLOBAL)> copy(::GlobalAlloc(GMEM_MOVEABLE, (len + 1) * sizeof(wchar_t)), ::GlobalFree);
+    std::unique_ptr<void, HGLOBAL(WINAPI*)(HGLOBAL)> copy(GlobalAlloc(GMEM_MOVEABLE, (len + 1) * sizeof(wchar_t)), GlobalFree);
     if (!copy)
         return;
 
-    if (!::OpenClipboard(nullptr))
+    HWND hWnd = GetActiveWindow();
+    if (!OpenClipboard(hWnd))
         return;
 
-    ::EmptyClipboard();
-
-    wchar_t* target = (wchar_t*)::GlobalLock(copy.get());
+    wchar_t* target = reinterpret_cast<wchar_t*>(GlobalLock(copy.get()));
+    if (target == nullptr) {
+        hsAssert(0, ST::format("GlobalLock() failed:\n{}", hsCOMError(hsLastWin32Error)).c_str());
+        return;
+    }
     memcpy(target, buf.data(), (len + 1) * sizeof(wchar_t));

```

Comment:
```suggestion
    memcpy(target, buf.data(), len * sizeof(wchar_t));
```
We explicitly initialize `target[len]` below, so no need to write it twice here...